src/timer/timer.rs

```diff
@@ -283,12 +287,10 @@ impl<MutexType: RawMutex> GenericTimerService<MutexType> {
     /// Returns a deadline based on the current timestamp plus the given Duration
     fn deadline_from_now(&self, duration: Duration) -> u64 {
         // TODO: Make this more efficient and with better overflow checking
```

Comment:
I think we can remove this comment with that change. I didn't think about `saturating_add` back then.